File: backend/internal/domain/school.go
package domain

// School คือข้อมูลโรงเรียน (สำหรับหน้าตั้งค่าระบบ)
type School struct {
	ID           string
	Name         string
	Code         string
	Address      Address // ที่อยู่แยกฟิลด์ (มาตรฐานไทย)
	Phone        string
	Email        string
	Website      string
	DirectorName string
	IsActive     bool
	// AttendanceLateAfter = เวลาตัดมา/สาย "HH:MM" (โซนไทย) — ตั้งค่าได้ต่อโรงเรียน
	AttendanceLateAfter string
	// AttendanceLatePenalty = คะแนนความประพฤติที่หักเมื่อมาสาย (0 = ไม่หัก)
	AttendanceLatePenalty int
}

// UpdateSchool payload แก้ไขข้อมูลโรงเรียน (code เป็นตัวระบุ ไม่ให้แก้)
type UpdateSchool struct {
	Name                  string
	Address               Address
	Phone                 string
	Email                 string
	Website               string
	DirectorName          string
	AttendanceLateAfter   string
	AttendanceLatePenalty int
}

File: frontend/src/shared/schemas/school.ts
import { z } from "zod";

import { addressSchema, type AddressData } from "./personnel";

// ข้อมูลโรงเรียน — ตรงกับ backend service.SchoolDTO
export const schoolSchema = z.object({
  id: z.string(),
  name: z.string(),
  code: z.string(),
  address: addressSchema,
  phone: z.string(),
  email: z.string(),
  website: z.string(),
  director_name: z.string(),
  is_active: z.boolean(),
  attendance_late_after: z.string(),
  attendance_late_penalty: z.number(),
});
export type School = z.infer<typeof schoolSchema>;

export type UpdateSchoolBody = {
  name: string;
  address: AddressData;
  phone: string;
  email: string;
  website: string;
  director_name: string;
  attendance_late_after: string;
  attendance_late_penalty: number;
};
